src/LiveComponent/src/LiveDownloadResponse.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Symfony\UX\LiveComponent;
+
+use SplFileInfo;
+use SplTempFileObject;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Symfony\Component\HttpFoundation\HeaderUtils;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ */
+final class LiveDownloadResponse extends BinaryFileResponse
+{
+    public const HEADER_LIVE_DOWNLOAD = 'X-Live-Download';
+
+    public function __construct(string|SplFileInfo $file, ?string $filename = null)
+    {
+        if (\is_string($file)) {
+            $file = new SplFileInfo($file);
+        }
+
+        if ((!$file instanceof SplFileInfo)) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not exist.', $file));
+        }
+
+        if ($file instanceof SplTempFileObject) {
+            $file->rewind();
+        }
+
+        parent::__construct($file, 200, [
+            self::HEADER_LIVE_DOWNLOAD => 1,
+            'Content-Disposition' => HeaderUtils::makeDisposition(HeaderUtils::DISPOSITION_ATTACHMENT, $filename ?? basename($file)),
+            'Content-Type' => 'application/octet-stream',
```

Comment:
Shouldn't we use the `$file`'s mime type?

Reply:
I hesitated (keep dep light) and chose not to...

But in fact it's a good idea and Mime is a very small component.

So let's make it a requirement for LiveComponent ? Or only when using downloads / uploads ? 

wdyt ?

Reply:
We can put it in `suggest` `composer.json` section (I don't remember what is the rule in Symfony, cc @nicolas-grekas), and then runtime check if `symfony/mime` exists (when 
 using downloads/uploads)

Reply:
Pretty sure there is no suggest in Symfony composer packages..

We have other situations in UX where we do not require a package (i.e. in Autocomplete for Form ..)

My question was more: should we require it for everyone, or keep it as an "runtime" dependency ?

Reply:
runtime dependency

Reply:
Why aren't you deferring to BinaryFileResponse for this logic?

Reply:
What's the reason for the header?

In the frontend, if not a standard/known content type, but has a content disposition (and an other requirements), trigger the download? This way we could say "to enable file downloads, return a response that has a content disposition"